### ISSUE TITLE
Style/RedundantSelf false negative

### ISSUE THREAD

**Issue by lucthev** [rubocop/rubocop#15155]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Cop name

Style/RedundantSelf

### Steps to reproduce the problem

```rb
# test.rb
def foo
  self.x ||= 1
end

def bar
  self.x
end
```

```
$ rubocop --only Style/RedundantSelf test.rb
```

### Expected behavior

The `self` within `def bar` is redundant, and `Style/RedundantSelf` should flag it as such.

### Actual behavior

No offenses detected.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.86.1 (using Parser 3.3.10.0, Prism 1.9.0, default, rubocop-ast 1.49.1, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin23]
```
